### ISSUE TITLE
Async file handling

### ISSUE THREAD

**Issue by aesteve** [FroMage/redpipe#34]
When looking at https://github.com/FroMage/redpipe/blob/master/redpipe-engine/src/main/java/net/redpipe/engine/resteasy/FileResource.java#L19

It seems that file handling is directly delegated to RestEasy. Not sure how it's implemented in details, but I'm concerned by the fact it might use blocking IO.

We could consider using Vert.x build-in ` FileSystem` which exposes some utilities regarding asynchronous IO.

**Comment by FroMage** [comment]
Yes, very good idea. We'd just need to do the same as RESTEasy does wrt preconditions and caching.
